server/controllers/follow.controllers.js

```diff
@@ -0,0 +1,118 @@
+import model from '../models';
+import validations from '../helpers/validations';
+
+const { User, Follower } = model;
+
+/**
+ * @description Controller to authenticate users
+ */
+const followController = {
+  async followUser(req, res) {
+    try {
+      const token = validations.verifyAuthHeader(req);
+      const { id: followerId } = token.userObj;
+      const { followeeId } = req.params;
+      // Get followee from followee table
+      const findUnFollowee = await User.findOne({ where: { id: followeeId } });
+      if (!findUnFollowee) {
+        res.status(404).json({
+          status: 404,
+          message: 'User not found',
+        });
+      }
+      if (findUnFollowee.id === followerId) {
+        res.status(406).json({
+          status: 406,
+          message: 'You cant follow yourself',
+        });
+      }
+      // Is follower already following followee?
+      const getllFollowerRecord = await Follower.findAll();
+      const userFollowing = getllFollowerRecord
+        .map(follower => follower)
+        .filter(
+          followed =>
+            followed.follower_id === followerId &&
+            followed.followee_id === followeeId
```

Comment:
Use a more meaningful variable name in the `map` function. Also what is the meaning of the variable `userFollowing`? @amaechi-chuks

Reply:
The variable userFollowing is a validation that checks for follower and followee in the database.

Reply:
> The variable userFollowing is a validation that checks for follower and followee in the database.

Can you describe what the variable is going to contain? Is it an object or array? And what will be its content(s)? @amaechi-chuks